src/VisualStudio.Roslyn.SDK/ComponentDebugger/DebugProfileProvider.cs

```diff
@@ -0,0 +1,95 @@
+
+        public Task OnBeforeLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile) => Task.CompletedTask;
+
+        public bool SupportsProfile(ILaunchProfile? profile) => Constants.CommandName.Equals(profile?.CommandName, StringComparison.OrdinalIgnoreCase);
+
+        public async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions, ILaunchProfile? profile)
+        {
+            // set up the managed (net fx) debugger to start a process
+            var settings = new DebugLaunchSettings(launchOptions)
+            {
+                LaunchDebugEngineGuid = Microsoft.VisualStudio.ProjectSystem.Debug.DebuggerEngines.ManagedOnlyEngine,
+                LaunchOperation = DebugLaunchOperation.CreateProcess
+            };
+
+            // try and get the target project
+            if (TryGetTargetProject(_configuredProject, profile, out var targetProjectUnconfigured))
+            {
+                settings.CurrentDirectory = Path.GetDirectoryName(targetProjectUnconfigured!.FullPath);
+                var compiler = _configuredProject.Capabilities.Contains(ProjectCapabilities.VB) ? "vbc.exe" : "csc.exe";
+                settings.Executable = Path.Combine(_compilerRoot, compiler);
+
+                // try and get the configured version of the target project
+                var targetProject = await targetProjectUnconfigured.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
+                if (targetProject is object)
+                {
+                    // get its compilation args
+                    var args = await targetProject.GetCompilationArgumentsAsync().ConfigureAwait(false);
+
+                    // append the command line args to the debugger launch
+                    settings.Arguments = string.Join(" ", args);
+                }
+            }
+
+            //PROTOTYPE: we probably shouldn't return anything when we couldn't figure it out
+            return new IDebugLaunchSettings[] { settings };
+        }
+
+        private static string GetCompilerRoot(SVsServiceProvider? serviceProvider)
+        {
+            // PROTOTYPE: we should try and work out the compiler location from the project itself
+            object rootDir = string.Empty;
+            var shell = (IVsShell?)serviceProvider?.GetService(typeof(SVsShell));
+            shell?.GetProperty((int)__VSSPROPID2.VSSPROPID_InstallRootDir, out rootDir);
+            return Path.Combine((string)rootDir, "MSBuild", "Current", "Bin", "Roslyn");
+        }
+
+        private static bool TryGetTargetProject(ConfiguredProject project, ILaunchProfile? profile, out UnconfiguredProject? targetProject)
+        {
+            var targetProjectPath = profile?.OtherSettings?.ContainsKey(Constants.TargetProjectPropertyName) == true
+                                    ? profile.OtherSettings[Constants.TargetProjectPropertyName].ToString()
+                                    : string.Empty;
```

Comment:
Hmm, is there some variant of `TryGetValue` on `OtherSettings` here?

Reply:
No, but I added another method to make the logic clearer

---
In reply to: [573354575](https://github.com/dotnet/roslyn-sdk/pull/726#discussion_r573354575) [](ancestors = 573354575)

Reply:
Apparently I just didn't find it before. There is. And i'm now using it.